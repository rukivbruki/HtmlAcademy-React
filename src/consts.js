export const AUTH = `AUTH`;
export const NO_AUTH = `NO_AUTH`;
