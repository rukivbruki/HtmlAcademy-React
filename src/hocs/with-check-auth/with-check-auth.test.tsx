import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {AUTH} from '../../consts.js';
import {BrowserRouter} from "react-router-dom";
import withCheckAuth from './with-check-auth';
import NameSpace from "../../reducer/name-space.js";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

interface MockComponentProps {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockComponentWrapped = withCheckAuth(MockComponent);

it(`withCheckAuth renders correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AUTH,
    }
  });

  const tree = renderer.create((
    <Provider store={store}>
      <BrowserRouter>
        <MockComponentWrapped/>
      </BrowserRouter>
    </Provider>

  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
