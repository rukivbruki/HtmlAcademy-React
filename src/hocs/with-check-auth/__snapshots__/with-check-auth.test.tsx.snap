// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withCheckAuth renders correctly 1`] = `<div />`;
